### PR TITLE
Perform check for server nonce if the auth redirection was triggered

### PR DESCRIPTION
**Pull request by sowmiya**
by the client app

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @sowmiya has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m [hunk 1]
```diff
@@ -64,14 +64,17 @@ - (void)performAuthorizationCodeGrantWithReceivedURL:(NSURL *)URL withCompletion
     NSString *authorizationCode = [URLQueryParams valueForKey:BOXAuthURLParameterAuthorizationCodeKey];
     NSString *authorizationError = [URLQueryParams valueForKey:BOXAuthURLParameterErrorCodeKey];
 
-    if ((_nonce || serverNonce) && [serverNonce isEqualToString:_nonce] == NO) {
-        NSError *error = [[NSError alloc] initWithDomain:BOXContentSDKErrorDomain 
-                                                    code:BOXContentSDKAuthErrorAccessTokenNonceMismatch
-                                                userInfo:nil];
-        if (block) {
-            block(nil, error);
+    NSString *authenticationRedirectURIScheme = [[NSURL URLWithString:self.redirectURIString] scheme];
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m:72]: Ah, good catch.

So if this is the check to see if we SHOULD/NEED TO do a nonce-comparison, is it better (more inclusive) to check if the server response included the 'state' query parameter (not if there is a value, if the key is there)?
We can double check if that's the better check and if it is, either do it in this commit or a separate one.

**Comment by boxtcarpel** [comment]
+2 LGTM

One comment

:green_book:

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m (hunk 1 above)
Comment (sowmiya) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m:72]: From what I checked with the server team, we are not 100% confident that the server will always return a state param(when it is empty) so we cannot base it on that. We will go with the current fix.

**Pull request merged**